app/models/user.rb

```diff
@@ -1,6 +1,21 @@
 class User < ApplicationRecord
-  has_many :votes
+  has_many :votes, dependent: :destroy
   has_many :ranked_works, through: :votes, source: :work
 
   validates :username, uniqueness: true, presence: true
+
+  def self.build_from_github(auth_hash)
+
+    user_data = {
+      uid: auth_hash[:uid],
+      username: auth_hash[:info][:username],
```

Comment:
Running this code resulted in a validation failure that the username was blank. I think the field Github sends is `nickname`, that's probably what you want on line 11.